test/unit/mock/slot-overwrite.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { MockContract, MockContractFactory, smock } from '@src';
+import { ADDRESS_EXAMPLE, ADDRESS_EXAMPLE2, ADDRESS_EXAMPLE3 } from '@test-utils';
+import { Bug, Bug__factory } from '@typechained';
+import { expect } from 'chai';
+
+describe('Mock: Slot Overwrite', () => {
+  let bugFactory: MockContractFactory<Bug__factory>;
+  let mock: MockContract<Bug>;
+
+  before(async () => {
+    bugFactory = await smock.mock('Bug');
+  });
+
+  beforeEach(async () => {
+    mock = await bugFactory.deploy();
+  });
+
+  describe('setVariable', () => {
+    it('should not be able to overwrite slot', async () => {
+      await mock.setVariable('myAddress', ADDRESS_EXAMPLE);
```

Comment:
No need to use any ADDRESS_EXAMPLEX, let just use generated random addresses, I believe we can achieve that with `makeRandomAddress()`